src/freon.c

```diff
@@ -152,24 +154,44 @@ int ioctl(int fd, unsigned long int request, ...) {
     return ret;
 }
 
+static int _open(int (*origfunc)(const char *pathname, int flags, mode_t mode),
```

Comment:
The intention was to preserve the forward reading of the original function across the split, which also happens to result in a smaller diff than placing the definition here.  That said, I'm happy with either way, so I've updated the pull request with the definition here (and the return fix).